onmt/utils/parse.py

```diff
@@ -111,15 +110,19 @@ def validate_train_opts(cls, opt):
             raise AssertionError(
                   "-gpu_ranks should have master(=0) rank "
                   "unless -world_size is greater than len(gpu_ranks).")
-        assert len(opt.data_ids) == len(opt.data_weights), \
-            "Please check -data_ids and -data_weights options!"
+        if hasattr(opt, 'data_ids'):
+            assert len(opt.data_ids) == len(opt.data_weights), \
```

Comment:
Is this necessary?

Reply:
No. For the opts part, it indeed needs cleaning.